### PR TITLE
add room struct with Message Adapter

### PR DESCRIPTION
**Pull request by jz709u**

### CODE REVIEW

File: CoThings/ViewController.swift
Comment (furkanmustafa) [inline CoThings/ViewController.swift]: This class sounds like it's better to be in a separate file. At least not in ViewController.

**Review by furkanmustafa — CHANGES_REQUESTED** [review]

File: CoThings/ViewController.swift
Comment (furkanmustafa) [inline CoThings/ViewController.swift]: Didn't check the whole code, but.. Do we have any way of error handling instead of ignoring? (discussion)

File: CoThings/ViewController.swift
Comment (furkanmustafa) [inline CoThings/ViewController.swift]: Feels like it's better to let it throw, and handle the error on the caller side.

Reason: the input provided here is obviously wrong/faulty. In such cases, throwing an exception is better IMO.

File: CoThings/ViewController.swift
Comment (jz709u) [reply CoThings/ViewController.swift]: yeah I can do that

Comment (furkanmustafa) [reply CoThings/ViewController.swift]: Thanks a lot!

File: CoThings/ViewController.swift
Comment (nesimtunc) [reply CoThings/ViewController.swift]: We don't have error handling mechanism yet, but better to have at the beginning, I'll also ask kindly to designer friends to show error cases in the app.

File: CoThings/ViewController.swift
Comment (jz709u) [reply CoThings/ViewController.swift]: if we are throwing an error I think we probably be throwing a different error instead of the one thrown by decoding error.  like conversion error of some sort

Comment (furkanmustafa) [reply CoThings/ViewController.swift]: I agree, that makes more sense.

File: CoThings/ViewController.swift
Comment (furkanmustafa) [reply CoThings/ViewController.swift]: Until we have a general flow of handling errors, we can just silence it here, and create a new issue for handling errors.

In that case it's better to leave a TODO comment here, and/or an issue number if possible. This would also allow merging this work without getting blocked on this.

**Comment by jz709u** [comment]
wrapped the swift pheonix into an adapter class, reducing direct dependencies on that library.  please check

File: CoThings/ViewController.swift
Comment (furkanmustafa) [reply CoThings/ViewController.swift]: Going to be handled in this issue => https://github.com/rainlab-inc/cothings-ios/issues/2
resolving.

File: CoThings/ViewController.swift
Comment (furkanmustafa) [reply CoThings/ViewController.swift]: I think last update also covers this, resolving!

**Comment by furkanmustafa** [comment]
Thanks @jz709u !

I don't understand swift much, so all my feedbacks were those three, and all three handled :)

@nesimtunc should review and merge.

**Review by furkanmustafa — APPROVED** [review]

**Comment by nesimtunc** [comment]
looks ✨ I just wonder why _Swift Developers_ don't use plural names for folders. Ex: ~~Models~~ Model :)

**Pull request merged**
